Reply/app/src/main/java/com/materialstudies/reply/data/Email.kt

```diff
@@ -33,14 +34,32 @@ data class Email(
     var isStarred: Boolean = false,
     var mailbox: Mailbox = Mailbox.INBOX,
     var timeSent: String = ""
-) {
+) : BaseObservable() {
     val hasBody: Boolean = body.isNotBlank()
     val hasAttachments: Boolean = attachments.isNotEmpty()
-    val recipientsPreview: String = recipients
-        .map { it.firstName }
-        .fold("") { name, acc -> "$acc, $name" }
+    val recipientsPreview: String = getRecipientPreview()
     val nonUserAccountRecipients = recipients
         .filterNot { AccountStore.isUserAccount(it.uid) }
+
+    private fun getRecipientPreview(): String {
```

Comment:
How about we modify what used to be here? Something along the lines of

```
    val recipientsPreview: String = recipients
            .map { it.firstName }
            .foldIndexed("") { i, acc, name ->
                val separator = when (i) {
                    0 -> ""
                    recipients.size - 1 -> " & "
                    else -> ", "
                }
                "$acc$separator$name"
            }
```

Reply:
Sounds, good didn't know a "kotliny" way to do it.